rpcs3/Emu/RSX/VK/vkutils/scratch.cpp

```diff
@@ -130,29 +145,34 @@ namespace vk
 
 	vk::buffer* get_scratch_buffer(u32 min_required_size)
 	{
-		if (g_scratch_buffer && g_scratch_buffer->size() < min_required_size)
+		auto& scratch_buffer = g_scratch_buffers_pool[0 /*TODO: Replace with Queue Family ID*/].get_buf();
+
+		if (scratch_buffer && scratch_buffer->size() < min_required_size)
 		{
 			// Scratch heap cannot fit requirements. Discard it and allocate a new one.
-			vk::get_resource_manager()->dispose(g_scratch_buffer);
+			vk::get_resource_manager()->dispose(scratch_buffer);
 		}
 
-		if (!g_scratch_buffer)
+		if (!scratch_buffer)
 		{
 			// Choose optimal size
 			const u64 alloc_size = std::max<u64>(64 * 0x100000, utils::align(min_required_size, 0x100000));
 
-			g_scratch_buffer = std::make_unique<vk::buffer>(*g_render_device, alloc_size,
+			auto new_buffer = new vk::buffer(*g_render_device, alloc_size,

```

Comment:
I don't think `new` can fail, there seems nothing wrong with make_unique either.
Perhaps if scratch buffer allocation fails, it signals in some other way.

Reply:
It would crash the driver anyway, I reverted back to make_unique. The 'new' thing was part of a crash debug exercise where I thought there was an issue with the move assignment but it was caused by something else.